### DISCUSSION TITLE
Is `it` a special variable now (in 0.92)?

### DISCUSSION THREAD

**Discussion by wdv4758h** [nushell/nushell#12440] [Q&A]
With 0.92, it seems that `it` is treated differently.

This print 42 in 0.91, but gets error in 0.92.

```nu
let tmp = {
  let it = 42
  print $it
}

do -c $tmp
```

errors in 0.92:

```sh
  × Missing parameter: $it.
   ╭─[example.nu:6:1]
 5 │ 
 6 │ do -c $tmp
   · ─┬
   ·  ╰── missing parameter: $it
   ╰────
```

This print 42 in both 0.91 and 0.92.

```nu
let tmp = {
  let x = 42
  print $x
}

do -c $tmp
```

**Comment by fdncred** [comment]
Hmmm, I don't think so. Could be a bug.

**Comment by ghost** [comment]
the only special thing i can think of for the `it` variable is as `iterated item` for inline loops (or however this is called):

```nu
ls | where ($it.name | str contains "foo")
```

but afaik this is just assigned within the `where` and shouldn't affect anything else..

---

i suspect that this issue could be due to it not recognising the variable as closure since the following (with explicit closure-markings) works:

```nu
let tmp = {||
  let it = 42
  print $it
}
do -c $tmp
```

EDIT: it recognises both as `closure` (`$tmp | describe`), but maybe something internal is different

**Reply by fdncred** [reply]
The `where` command has a "SyntaxShape::RowCondition" which reveals itself with `it`. No other command has a RowCondition. We though it was too much magic but it fits with `where`.

**Reply by sholderbach** [reply]
Thanks for the investigation @Jan9103 !
